lib/resty/core/request.lua

```diff
@@ -332,6 +373,19 @@ do
 
         error("error")
     end
+
+    _M.set_req_header = set_req_header
+
+    local orig_func = ngx.req.set_header
+
+    function ngx.req.set_header(name, value)
+        if type(value) == "table" then
+            return orig_func(name, value)
+        end
+
+        local override = true
```

Comment:
Better to avoid using a variable here and directly pass the argument. A comment can explain what it is if desired.